checkpoint/ddb/ddb_test.go

```diff
@@ -0,0 +1,105 @@
+package ddb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+type fakeRetryer struct {
+	Name string
+}
+
+func (r *fakeRetryer) ShouldRetry(err error) bool {
+	r.Name = "fakeRetryer"
+	return false
+}
+
+func TestNewCheckpoint(t *testing.T) {
+	c, err := New("", "")
+	if c == nil {
+		t.Errorf("expected checkpoint client instance. got %v", c)
+	}
+	if err != nil {
+		t.Errorf("new checkpoint error expected nil. got %v", err)
+	}
+}
+
+func TestCheckpointSetting(t *testing.T) {
+	var ck Checkpoint
+	ckPtr := &ck
+
+	// Test WithMaxInterval
+	setInterval := WithMaxInterval(time.Duration(2 * time.Minute))
+	setInterval(ckPtr)
+
+	// Test WithRetryer
+	var r fakeRetryer
+	setRetryer := WithRetryer(&r)
+	setRetryer(ckPtr)
+
+	// Test WithDyanmoDBClient
+	var fakeDbClient = dynamodb.New(
+		session.New(aws.NewConfig()), &aws.Config{
+			Region: aws.String("us-west-2"),
+		},
+	)
+	setDDBClient := WithDynamoClient(fakeDbClient)
+	setDDBClient(ckPtr)
+
+	// Validation
```

Comment:
I don't think the comment is needed since your code clearly explains what are you doing here. I would suggest removing the comment.